plugin/vm/VM.ml

```diff
@@ -83,54 +70,104 @@ let k_to_mod_acct (acct: k) : modified_account = match acct with
 | _ -> failwith "Invalid value where account was expected"
 
 let k_to_log (log: k) : log_entry = match log with
-  [KApply3(LbllogEntry, [Int address], [List(SortList,Lbl_List_,topics)], [String data])] ->
+  [KApply3(LbllogEntry, [Int address], [List(SortList,Lbl_List_,topics)], data)] ->
   let z_topics = List.map k_to_z topics in
-  {address=World.of_z_width 20 address;topics=List.map (World.of_z_width 32) z_topics;data=Bytes.of_string data}
+  let module Def = (val Plugin.get ()) in
+  let bytes_data = 
+    match Def.eval (KApply(LblunparseByteStack, [data])) [Bottom] with
+    | [String bytes] -> Bytes.of_string bytes
+    | _ -> failwith "Invalid value where string was expected"
+  in
+  {address=World.of_z_width 20 address;topics=List.map (World.of_z_width 32) z_topics;data=bytes_data}
 | _ -> failwith "Invalid value found where SubstateLogEntry was expected"
 
 let z_of_rlp rlp =
   match rlp with
   Rlp.RlpData rope -> World.to_z (Bytes.of_string (Rope.to_string rope))
 | Rlp.RlpList _ -> failwith "Invalid value where rlp-encoded string expected"
 
-let unpack_input iscreate data code =
-  let rlp = Rlp.decode (Rope.of_string (Bytes.to_string data)) in
-  match rlp with
-  Rlp.RlpList[Rlp.RlpData(rope);Rlp.RlpList(args)] ->
-  let z_args = List.map z_of_rlp args in
-  let str = Rope.to_string rope in
-  if iscreate then
-    str,z_args,""
-  else
-    Bytes.to_string code,z_args,str
-| _ -> failwith "Invalid value where rlp-encoded args and function name expected"
+let unpack_input iscreate data code config =
+  match config with
+  | Iele_config -> 
+  begin
+    let rlp = Rlp.decode (Rope.of_string (Bytes.to_string data)) in
+    match rlp with
+    Rlp.RlpList[Rlp.RlpData(rope);Rlp.RlpList(args)] ->
+    let z_args = List.map z_of_rlp args in
+    let k_args = List.map (fun z -> [Int z]) z_args in
+    let args_list = [List(SortList,Lbl_List_,k_args)] in
+    let str = Rope.to_string rope in
+    if iscreate then
+      str,args_list,""
+    else
+      Bytes.to_string code,args_list,str
+    | _ -> failwith "Invalid value where rlp-encoded args and function name expected"
+  end
+  | Ethereum_config _ ->
+  begin
+    let args = [String(Bytes.to_string data)] in
+    if iscreate then
+      Bytes.to_string data,args,""
+    else
+      Bytes.to_string code,args,""
+  end
 
 let pack_output rets =
   let l = List.map (fun ret -> Rlp.RlpData (Rope.of_string (Bytes.to_string ret))) rets in
   let rlp = Rlp.RlpList l in
   Bytes.of_string (Rope.to_string (Rlp.encode rlp))
 
+let get_schedule blocknum config = match config with
```

Comment:
It looks like you are trying to handle the differences between EVM and IELE by writing code that takes a flag, and being careful that the unused branch will at least compile against the `constants.ml` of the other definition.

I suppose it's okay for now, but we can't count on that in the long term. Instead, this code could be wrapped into a functor to be instantiated with definition-specific implementations when it's used from the IELE and EVM semantics.